tests/ui/type-alias-impl-trait/self-referential-in-fn-trait.rs

```diff
@@ -0,0 +1,10 @@
+#![feature(type_alias_impl_trait)]
```

Comment:
how does this behave on the new trait solver? I think the new trait solver does re-prove bounds even if it's unifying with an existing opaque:

https://github.com/rust-lang/rust/blob/bd9785cce8dffad58a793f85454a5064100e9c84/compiler/rustc_trait_selection/src/solve/eval_ctxt.rs#L896-L902

Reply:
it works (see new commit), but I haven't run any logs yet to see what it is doing, and whether anything is different.

Reply:
Wonder why it's not cycling. I bet it's because of lazy norm. Anyways, I guess it shouldn't matter, unless we make sure we don't commit to something that breaks the new solver.